src/core/labeling/qgsvectorlayerlabelprovider.h

```diff
@@ -22,6 +22,7 @@
 #include "qgslabelingengine.h"
 #include "qgsrenderer.h"
 #include "qgstextrenderer.h"
+#include "qgspallabeling.h"
```

Comment:
Same here, required? (I'm starting to assume so seeing it twice :) )

Reply:
Yep, same reason. It's actually a tightening of includes and more heavy use of forward declrations.